src/libraries/System.Security.Cryptography.Cose/tests/CoseTestHelpers.cs

```diff
@@ -373,6 +501,26 @@ private static void AssertHeaders(CborReader reader, List<(CoseHeaderLabel, Read
         internal static RSA RSAKey => t_rsaKey ??= CreateRSA(true);
         internal static RSA RSAKeyWithoutPrivateKey => t_rsaKeyWithoutPrivateKey ??= CreateRSA(false);
 
+        [ThreadStatic]
+        private static MLDsa? t_mldsa44Key;
+        [ThreadStatic]
+        private static MLDsa? t_mldsa44KeyWithoutPrivateKey;
+        [ThreadStatic]
+        private static MLDsa? t_mldsa65Key;
+        [ThreadStatic]
+        private static MLDsa? t_mldsa65KeyWithoutPrivateKey;
+        [ThreadStatic]
+        private static MLDsa? t_mldsa87Key;
+        [ThreadStatic]
+        private static MLDsa? t_mldsa87KeyWithoutPrivateKey;
```

Comment:
Consider, as a followup, introducing a private nested class to hold all of these ThreadStatic values, e.g.

```C#
[ThreadStatic]
private static KeyRing t_threadKeys;
private static KeyRing ThreadKeys => (t_threadKeys ??= new KeyRing());

internal static MLDsa MLDsa44Key => ThreadKeys.MLDsa44Key;

...

private sealed class KeyRing
{
    internal MLDsa MLDsa44 { get => field ??= CreateMLDsa(MLDsaAlgorithm.MLDsa44, true); private set; }
    ...
}
```

It's not that what's here is bad, it's just that it's a lot of individual ThreadStatic values, and over time it just feels like it's increasing the risk of missing one and having parallelism where we don't expect it.